terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -265,7 +270,7 @@ def check_required_files
       end
 
       def hashes_object_regex
-        /hashes\s*=\s*.*\]/m
+        /hashes\s*=\s*[^\]]*\]/m
```

Comment:
This makes the regex less greedy so it will stop within each provider block and not match over all provider blocks at the same time.